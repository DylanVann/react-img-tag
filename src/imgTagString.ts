export interface ImgTagProps {
    class?: string
    className?: string
    src?: string
    srcSet?: string
    sizes?: string
    title?: string
    alt?: string
    width?: number
    height?: number
}

export const imgTagString = (props: ImgTagProps = {}): string => {
    const classOrClassName = props.className || props.class
    const classString = classOrClassName ? ` class="${classOrClassName}"` : ``
    const src = props.src ? `src="${props.src}" ` : `src="" ` // required attribute
    const srcSet = props.srcSet ? `srcset="${props.srcSet}" ` : ``
    const sizes = props.sizes ? `sizes="${props.sizes}" ` : ``
    const title = props.title ? `title="${props.title}" ` : ``
    const alt = props.alt ? `alt="${props.alt}" ` : `alt="" ` // required attribute
    const width = props.width ? `width="${props.width}" ` : ``
    const height = props.height ? `height="${props.height}" ` : ``

    return `<img ${width}${height}${src}${srcSet}${alt}${title}${sizes}${classString}/>`
}
