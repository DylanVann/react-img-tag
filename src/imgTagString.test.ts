import { imgTagString } from "./imgTagString";

it('outputs a string', () => {
    expect(imgTagString()).toMatchSnapshot()
})

it('outputs a string with props', () => {
    expect(imgTagString({
        class: 'test-class',
        src: 'image.png',
        srcSet: 'image.png',
        sizes: 'the-sizes',
        title: 'title',
        alt: 'a description',
        width: 20,
        height: 10,
    })).toMatchSnapshot()
})