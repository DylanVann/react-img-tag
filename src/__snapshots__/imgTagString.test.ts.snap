// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs a string 1`] = `"<img src=\\"\\" alt=\\"\\" />"`;

exports[`outputs a string with props 1`] = `"<img width=\\"20\\" height=\\"10\\" src=\\"image.png\\" srcset=\\"image.png\\" alt=\\"a description\\" title=\\"title\\" sizes=\\"the-sizes\\"  class=\\"test-class\\"/>"`;
